### ISSUE TITLE
Instant Search: faster mobile search refinements

### ISSUE THREAD

**Issue by gibrown** [Automattic/jetpack#14783]
There are a number of cases where I think we should auto scroll the page for the user to make it faster/easier to see the top search result and refine the search query:
- on mobile, after scrolling down the page (e.g. after a load more). Once the user starts scrolling up a little bit, let's show the search box and the filtering selections so that the user doesn't have to scroll up all the way. A lot of apps seem to do this, and I feel myself missing it.
- on mobile, after a filter is selected further down the list we should rescroll the user back to the top of the page so that we show the first search result. It feels odd to just have the dropdown disappear
- on desktop, the same thing... I think we should scroll to the top of the page if a filter that was further down the page was applied.

**Comment by keoshi** [comment]
> Once the user starts scrolling up a little bit, let's show the search box and the filtering selections

I agree this is useful, but it could be hard to get a perfect implementation of this. I haven't seen a really good one that isn't native — the ones I've seen on the web all have some sort of disadvantage or compromise. Plus, on mobile devices you can touch the top of the browser and it scrolls up automatically, so I don't think we absolutely _need_ this feature.

As for the other two, I completely agree. We should try to have a smooth scroll so it's not too jarring.
